base/deprecated.jl

```diff
@@ -331,7 +331,13 @@ function setproperty!(ci::CodeInfo, s::Symbol, v)
     return setfield!(ci, s, convert(fieldtype(CodeInfo, s), v))
 end
 
-@eval Threads nthreads() = threadpoolsize()
+@eval Threads function nthreads()
+    @warn """
+    In julia 1.9+ which supports threadpools (see [link to docs])
+    use of threads should be in the context of which threadpool is going to be used, so `nthreads()` should be
+    replaced with `nthreads(:default)`, or `nthreads(:interactive)` if interactive threads are requrested.""" maxlog = 1
```

Comment:
There is not one simple replacement, as many usages must be replaced with task_local_storage instead or dropped entirely. But the replacement is almost never `nthreads(:interactive)`
```suggestion
    replaced with `nthreads(:default)` or `task_local_storage`, as appropriate.
```

Reply:
Also, AFAIU, the "number of threads" is at least partially a dynamic concept now and potentially/likely(?) even more so in the future. That is, the result of `nthreads` (with or without argument) isn't necessarily constant (per Julia session) anymore. This is something that users would need to be informed/warned about as well.

Reply:
Updated with both suggestions